hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -280,117 +323,130 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
+                self._main_task = safe_ensure_future(self.main(timestamp))
+
+        for market_pair in self._market_pairs.values():
+            if self.is_gateway_market(market_pair.taker):
+                if self._cancel_outdated_orders_task is None or self._cancel_outdated_orders_task.done():
+                    self._cancel_outdated_orders_task = safe_ensure_future(self.apply_gateway_transaction_cancel_interval())
+
+    async def main(self, timestamp: float):
+        try:
             # Calculate a mapping from market pair to list of active limit orders on the market.
             market_pair_to_active_orders = defaultdict(list)
 
-            for maker_market, limit_order in active_limit_orders:
+            for maker_market, limit_order in self.active_limit_orders:
                 market_pair = self._market_pairs.get((maker_market, limit_order.trading_pair))
                 if market_pair is None:
                     self.log_with_clock(logging.WARNING,
                                         f"The in-flight maker order in for the trading pair '{limit_order.trading_pair}' "
                                         f"does not correspond to any whitelisted trading pairs. Skipping.")
                     continue
 
-                if not self._sb_order_tracker.c_has_in_flight_cancel(limit_order.client_order_id) and \
+                if not self._sb_order_tracker.has_in_flight_cancel(limit_order.client_order_id) and \
                         limit_order.client_order_id in self._maker_order_ids:
                     market_pair_to_active_orders[market_pair].append(limit_order)
 
             # Process each market pair independently.
             for market_pair in self._market_pairs.values():
-                self.c_process_market_pair(market_pair, market_pair_to_active_orders[market_pair])
+                await self.process_market_pair(timestamp, market_pair, market_pair_to_active_orders[market_pair])
+
             # log conversion rates every 5 minutes
-            if self._last_conv_rates_logged + (60. * 5) < self._current_timestamp:
+            if self._last_conv_rates_logged + (60. * 5) < timestamp:
                 self.log_conversion_rates()
-                self._last_conv_rates_logged = self._current_timestamp
+                self._last_conv_rates_logged = timestamp
         finally:
             self._last_timestamp = timestamp
 
-    def has_active_taker_order(self, object market_pair):
-        cdef dict market_orders = self._sb_order_tracker.c_get_market_orders()
+    async def get_gateway_quotes(self):
+        for market_pair in self._market_pairs.values():
+            if self.is_gateway_market(market_pair.taker):
+                _, _, quote_rate, _, _, base_rate = self.get_taker_to_maker_conversion_rate()
+                order_amount = self._order_amount * base_rate
+                market: GatewayEVMAMM = cast(GatewayEVMAMM, market_pair.taker.market)
```

Comment:
Do we really need to cast the value?

Reply:
Probably not but it was like that in `amm_arb` . 
I'll remove it.

Reply:
✅